src/Components/Publishing/Display/__test__/DisplayPanel.test.js

```diff
@@ -0,0 +1,135 @@
+      <DisplayPanel unit={UnitPanelVideo} campaign={Campaign} />
+    ).toJSON()
+
+    expect(displayPanel).toMatchSnapshot()
+  })
+})
+
+describe('units', () => {
+  const getWrapper = (props = {}) => {
+    return mount(
+      <DisplayPanel
+        unit={UnitPanel}
+        campaign={Campaign}
+        {...props}
+      />
+    )
+  }
+
+  describe('tracking', () => {
+    it('calls on mount', () => {
+      getWrapper()
+      expect(track).toHaveBeenCalled()
+    })
+
+    it('calls on click', () => {
+      const wrapper = getWrapper()
+      wrapper.simulate('click')
+      expect(track).toHaveBeenCalled()
+    })
+
+    it('calls on mouse enter', () => {
+      const wrapper = getWrapper()
+      wrapper.simulate('mouseEnter')
+      expect(track).toHaveBeenCalled()
+    })
+
+    it('calls on video click', () => {
+      const wrapper = getWrapper()
+      wrapper.simulate('mouseEnter')
+      const instance = wrapper.instance()
+      instance.isVideoClickArea = () => true
+      wrapper.find('DisplayPanelContainer').simulate('click')
+      expect(track).toHaveBeenCalled()
+    })
+  })
+
+  it('opens link on click', () => {
+    const wrapper = getWrapper()
+    const spy = jest.spyOn(wrapper.instance(), 'openLink')
+    wrapper.update()
+    wrapper.simulate('click')
+    expect(spy).toHaveBeenCalled()
+  })
+
+  describe('video', () => {
+    it('renders a video cover if not playing', () => {
+      const unit = clone(UnitPanel)
+      unit.assets[0].url = 'foo.mp4'
+      const wrapper = getWrapper({ unit })
+      expect(wrapper.find('VideoCover').length).toEqual(1)
+    })
+
+    it('hides the video cover when playing', () => {
+      const unit = clone(UnitPanel)
+      unit.assets[0].url = 'foo.mp4'
+      const wrapper = getWrapper({ unit, isMobile: true })
+      const instance = wrapper.instance()
+      instance.isVideoClickArea = () => true
+      wrapper.find('DisplayPanelContainer').simulate('click')
+      expect(wrapper.find('VideoCover').length).toEqual(0)
+    })
+
+    it('on mobile, renders mini video controls', () => {
+      const wrapper = getWrapper({ isMobile: true })
+      expect(wrapper.find('VideoControls').length).toEqual(1)
+    })
+
+    it('renders a <video />', () => {
+      const wrapper = getWrapper()
+      expect(wrapper.find('video').length).toEqual(1)
+    })
+
+    it('on mobile, plays video on click', () => {
+      const wrapper = getWrapper({ isMobile: true })
+      const instance = wrapper.instance()
+      instance.isVideoClickArea = () => true
+      instance.video = { play: jest.fn(), pause: jest.fn() }
+      const spy = jest.spyOn(instance, 'onClickVideo')
+      wrapper.update()
+      wrapper.find('DisplayPanelContainer').simulate('click')
+      expect(spy).toHaveBeenCalled()
+      expect(wrapper.state().isPlaying).toEqual(true)
+      wrapper.find('DisplayPanelContainer').simulate('click')
+      expect(wrapper.state().isPlaying).toEqual(false)
```

Comment:
@eessex - a lot of the testing things we talked about the other day are in this block. Unless anyone has any issues with this, note that I named the file `.js` to get around typescript errors related to stubbing / overriding methods.